### PR TITLE
Topological style provider

### PR DESCRIPTION
**Pull request by quinarygio**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ x] The commit message follows our guidelines
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#60 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
image

Color scale should be improved. In this substation with 4 buses we can see that second bus (from the right) is ok, but the 2 others (2 first) are kind of greyish but not a red variant.

**Comment by quinarygio** [comment]
> image
> 
> Color scale should be improved. In this substation with 4 buses we can see that second bus (from the right) is ok, but the 2 others (2 first) are kind of greyish but not a red variant.
> Anyway, we can merge this PR it as it is and let the issue open to rework on color scale generation. Is it ok for you?

Ok for me to merge the PR and let the issue open. 
To have a better color gradient we could define a min and max color for each base voltage and then it will be easier to calculate the required color steps.

File: substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.substationdiagram.library.ComponentType;
+import com.powsybl.substationdiagram.model.Edge;
+import com.powsybl.substationdiagram.model.Graph;
+import com.powsybl.substationdiagram.model.Node;
+import com.powsybl.substationdiagram.model.Node.NodeType;
+import com.powsybl.substationdiagram.svg.DefaultSubstationDiagramStyleProvider;
+import com.powsybl.substationdiagram.svg.SubstationDiagramStyles;
+
+import javafx.scene.paint.Color;
```
Comment (geofjamg) [inline substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java]: SVG generation is intended to be execute on server side and we not should be dependent on JavaFX on server side.
Could you replace javafx.scene.paint.Color by something else that does not depend on JavaFX ou Swing or AWT.
When we will switch to Java 9 module system, we won't add Java dependency on server side.

Comment (quinarygio) [reply substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java]: It's possible to compute  color gradient using only RGB values, but the stat and end colors must be specified.  I could add minColor and maxColor parameters to BaseVoltageColor configuration:

```
baseVoltages:
    - name: "225"
      minValue: 180
      maxValue: 300
      color: "#00FF00"
      minColor: "#49FA6C"
      maxColor: "#226304"
      profile: "TEST"
```

It is ok?

Comment (geofjamg) [reply substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java]: minColor and maxColor has nothing to do in base voltage config.
Why not a "delta" in config specific to substation diagram.
How your comment is related to JavaFX color import ?

Comment (quinarygio) [reply substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java]: As we cannot use JavaFX or AWT or Swing Color classse, I tried to work only with RGB codes. I do not know how to get different variants of a given color ensuring that it remains in the same color group ( N red color variants for example). Instead if we configure the min and max color as a ligh red and a dark red it is possible to compute N color variants in that range.

Comment (geofjamg) [reply substation-diagram/substation-diagram-core/src/main/java/com/powsybl/substationdiagram/util/TopologicalStyleProvider.java]: Did you search for a third party library non dependent to JavaFX which would have this kind of color scale generation?
Maybe we could try to port some javascript code in Java like this one: https://hihayk.github.io/scale/ ?

**Pull request merged**